PowerShellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingServiceUtilities.cs

```diff
@@ -145,28 +147,47 @@ private void InitializeRunspace(PSHost psHost)
             _runspace = RunspaceFactory.CreateRunspace(psHost, iss);
             _runspace.Open();
 
-            LoadProfile();
+            ProvideProfileVariable();
+            LoadProfiles();
             ServiceCommon.Log("Done initializing runspace");
         }
 
-        private void LoadProfile()
+        private void LoadProfiles()
         {
-            ServiceCommon.Log("Loading PowerShell Profile");
+            if (_callback == null)
+            {
+                _callback = OperationContext.Current.GetCallbackChannel<IDebugEngineCallback>();
+            }
+
+            if (!_callback.ShouldLoadProfiles())
+            {
+                return;
+            }
+
+            ServiceCommon.Log("Loading PowerShell Profiles");
             using (PowerShell ps = PowerShell.Create())
             {
-                var myDocuments = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
-                var windowsPowerShell = Path.Combine(myDocuments, "WindowsPowerShell");
-                var profile = Path.Combine(windowsPowerShell, "PoshTools_profile.ps1");
+                ps.Runspace = _runspace;
 
-                var fi = new FileInfo(profile);
-                if (!fi.Exists)
+                PSObject profiles = _runspace.SessionStateProxy.PSVariable.Get("profile").Value as PSObject;
+                foreach(PSMemberInfo member in profiles.Members)

```

Comment:
We don't have a guarantee that the only members on this variable are the profiles to load.

Also, we should deterministically load the profiles in the order mentioned in the original issue.